src/lgar.cxx

```diff
@@ -1728,19 +1866,35 @@ extern void lgar_fix_dry_over_wet_wetting_fronts(double *mass_change, double* cu
 	  // now this is the wetting front that was below the dry wetting front
 	  current->psi_cm = calc_h_from_Se(Se_k, vg_a_k, vg_m_k, vg_n_k);
 
-	  struct wetting_front *current_local = *head;
+	  struct wetting_front *current_local = *head; //shouldn't be head, should be the highest to_bottom WF above the one that got deleted before the next to_bottom==FALSE one 
+    current_local = listFindFront(current->front_num - 1, *head, NULL);
+    if (listFindFront(current_local->front_num - 1, *head, NULL)!=NULL){
+      while (listFindFront(current_local->front_num - 1, *head, NULL)->to_bottom==TRUE){
+        current_local = listFindFront(current_local->front_num - 1, *head, NULL);
+        if (listFindFront(current_local->front_num - 1, *head, NULL)==NULL){
+          break;
+        }
+      }
+    }
```

Comment:
The idea behind this code is that we don’t always want to correct the psi values all of the WFs in layers above the current one, which is what would happen if current_local is always set to head. This is because, we only want to correct the psi value of wetting fronts above the current WF before we hit the next to_bottom == FALSE wetting front (or until and including *head, if no to_bottom == FALSE wetting fronts exist above the current one). The reason for this is that the next highest to_bottom == FALSE wetting front was not part of the dry over wet merge. Now, it is almost always the case that, even with the code I added, the wetting front selected for current_local will be the head WF. The reason for this is that dry over wet merging overwhelmingly occurs for the top most wetting front that has to_bottom == FALSE, because this is the WF from which AET is extracted, and so there are no next to_bottom == FALSE wetting fronts above the current one. The code I wrote covers the very rare case where dry over wet merging happens below the WF that yields water for the AET demand. This case is very rare because it implies that a wetting front got both deeper and drier than the one below it in the same time step, which is highly unlikely but possible in LGAR. We can chat more about it if you want.

Reply:
Seems that *head works for this code in LGAR, so I'm deleting my added code here and we'll add some function that does something similar to the code I suggested here, in LGARTO